api/clientcredentialsloginprovider.go

```diff
@@ -45,6 +77,12 @@ func (p *clientCredentialsLoginProvider) AuthHeader() (http.Header, error) {
 // It authenticates as the entity using client credentials.
 // Subsequent requests on the state will act as that entity.
 func (p *clientCredentialsLoginProvider) Login(ctx context.Context, caller base.APICaller) (*LoginResultParams, error) {
+	if p.populatedFromEnvironment {
```

Comment:
Do we need this? We want `p.clientIdAndSecretSet()` regardless of how the provider is populated?